sycl/include/CL/sycl/buffer.hpp

```diff
@@ -126,6 +139,26 @@ class buffer {
         detail::getNextPowerOfTwo(sizeof(T)), propList);
   }
 
+  // This constructor is a prototype for a future SYCL specification
+  template <class Container, int N = dimensions,
+            typename = EnableIfOneDimension<N>,
+            typename = EnableIfContiguous<Container>>
+  buffer(Container &container, AllocatorT allocator,
+         const property_list &propList = {})
+      : Range(range<1>(container.size())) {
+    impl = std::make_shared<detail::buffer_impl<AllocatorT>>(
+        container.data(), container.data() + container.size(),
+        get_count() * sizeof(T), detail::getNextPowerOfTwo(sizeof(T)), propList,
+        allocator);
+  }
+
+  // This constructor is a prototype for a future SYCL specification
+  template <class Container, int N = dimensions,
```

Comment:
Suggest adding some macro here to avoid user relying on this feature to be available in all version of all SYCL implementation.

Reply:
I agree. The extension doesn't have a name yet. Should I use `__SYCL_INTEL_CONSTRUCT_FROM_CONTAINER__`? I got the feedback I should also add the equivalent for `image` so BUFFER_CONSTRUCTOR is likely not a good name. Also until we resolve https://github.com/intel/llvm/issues/806 there is no convenient way to enable it. Is it OK if I don't add the `#ifdef` in this PR and we do it when we resolved the issue and add it for all extensions? Should I add a TODO comment?

Reply:
Yes, it's OK if we resolve that in a separate patch.
Don't think TODO would be valuable here, but it's up to you.